gcovr/coveralls_generator.py

```diff
@@ -0,0 +1,197 @@
+            write_json(gcovr_json_dict, output)
+
+
+def print_coveralls_report(covdata, output_file, options):
+    """
+    Outputs a JSON report in the Coveralls API coverage format
+
+    @param covdata: is a dictionary of file coverage objects, keyed with an absolute filepath
+    @param output_file: is the name of the file to create
+    @param options: options object
+    """
+
+    # Create object to collect coverage data
+    json_dict = {}
+
+    # Capture timestamp
+    json_dict['run_at'] = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S UTC")
+
+    # Pull environment variables
+    if(os.environ.get('COVERALLS_REPO_TOKEN') is not None):
+        json_dict['repo_token'] = os.environ.get('COVERALLS_REPO_TOKEN')
+
+    CurrentBranch = None
+    CurrentCommit = None
+    # Consume Travis CI specific environment variables _(if available)_
+    # See https://docs.travis-ci.com/user/environment-variables
+    if (os.environ.get('TRAVIS_JOB_ID') is not None):
+        json_dict['service_name'] = "travis-ci"
+        json_dict['service_job_id'] = os.environ.get('TRAVIS_JOB_ID')
+        json_dict['service_number'] = os.environ.get('TRAVIS_BUILD_NUMBER')
+        json_dict['service_pull_request'] = os.environ.get('TRAVIS_PULL_REQUEST')
+        CurrentCommit = os.environ.get('TRAVIS_COMMIT')
+        CurrentBranch = os.environ.get('TRAVIS_BRANCH')
+    # Consume Appveyor specific environment variables _(if available)_
+    # See https://www.appveyor.com/docs/environment-variables/
+    elif (os.environ.get('APPVEYOR_URL') is not None):
+        json_dict['service_name'] = "appveyor"
+        json_dict['service_job_id'] = os.environ.get('APPVEYOR_JOB_ID')
+        json_dict['service_number'] = os.environ.get('APPVEYOR_JOB_NUMBER')
+        json_dict['service_pull_request'] = os.environ.get('APPVEYOR_PULL_REQUEST_NUMBER')
+        CurrentCommit = os.environ.get('APPVEYOR_REPO_COMMIT')
+        CurrentBranch = os.environ.get('APPVEYOR_REPO_BRANCH')
+    # Consume Jenkins specific environment variables _(if available)_
+    # See https://opensource.triology.de/jenkins/pipeline-syntax/globals
+    elif (os.environ.get('JENKINS_URL') is not None):
+        json_dict['service_name'] = "jenkins-ci"
+        json_dict['service_job_id'] = os.environ.get('JOB_NAME')
+        json_dict['service_number'] = os.environ.get('BUILD_ID')
+        json_dict['service_pull_request'] = os.environ.get('CHANGE_ID')
+        if (os.environ.get('GIT_COMMIT') is not None):
+            CurrentCommit = os.environ.get('GIT_COMMIT')
+        CurrentBranch = os.environ.get('BRANCH_NAME')
+    elif (os.environ.get('GCOVR_TEST_SUITE') is not None):
+        json_dict['service_name'] = "gcovr-test-suite"
+        json_dict['service_job_id'] = 'id'
+        json_dict['service_number'] = 'number'
+        json_dict['service_pull_request'] = 'pr'
+        CurrentCommit = None
+        CurrentBranch = 'branch'
+
+    git = shutil.which('git')
```

Comment:
Well, I don't know how useful it is but maybe is better to use something like `--gcov-executable` option when we can point out to particular `gcov` exacutable. We can also set `gcov` exacutable by `GCOV` variable. Maybe it is good to to the same for `git`.

Reply:
For me the option make sense for gcov. There can be installed several versions at the same time. But for git I don't know if this is an use case.
I can add the option if you think thats an use case.

Reply:
Ok, it was just proposal. I also can't find any practical example where it can be used, so leave it as it is.